src/hotspot/share/cds/lambdaFormInvokers.hpp

```diff
@@ -26,21 +26,28 @@
 #define SHARE_CDS_LAMBDAFORMINVOKERS_HPP
 #include "memory/allStatic.hpp"
 #include "runtime/handles.hpp"
+#include "utilities/growableArray.hpp"
 
-template <class T>
-class GrowableArray;
 class ClassFileStream;
+template <class T> class Array;
 
 class LambdaFormInvokers : public AllStatic {
  private:
-  static GrowableArray<char*>* _lambdaform_lines;
+  static GrowableArrayCHeap<char*, mtClassShared>* _lambdaform_lines;
+  // for storing LF form lines (LF_RESOLVE only) in read only table.
+  static Array<Array<char>*>* _static_archive_invokers;
   static void reload_class(char* name, ClassFileStream& st, TRAPS);
  public:
-
   static void append(char* line);
+  static void append_filtered(char* line);
+  // All exceptions except for OOM will be suppressed in the fucntion, the
+  // non-OOM failure should not affect normal archiving.
```

Comment:
The above comments should be removed, since we do throw exceptions if the generated classfiles are invalid.

Reply:
Will update, thanks!